QUESTION 1:What is the northernmost airport in the United States?

In [14]:
import pandas as pd

Upload airports.csv

In [16]:
airports = pd.read_csv("airports.csv")

Locate the northernmost airport in the United States

In [17]:
northernmost = us_airports.sort_values(by="lat", ascending=False)
northernmost.head()

,faa,name,lat,lon,alt,tz,dst,tzone
230,BRW,Wiley Post Will Rogers Mem,71.285446,-156.766003,44,-9,A,America/Anchorage
110,AIN,Wainwright Airport,70.638056,-159.994722,41,-9,A,America/Anchorage
708,K03,Wainwright As,70.613378,-159.860350,35,-9,A,America/Anchorage
152,ATK,Atqasuk Edward Burnell Sr Memorial Airport,70.467300,-157.436000,96,-9,A,America/Anchorage
1363,UUK,Ugnu-Kuparuk Airport,70.330833,-149.597500,67,-9,A,America/Anchorage


After filtering the dataset to include only U.S. airports and sorting by latitude in descending order, I determined that Wiley Post–Will Rogers Memorial Airport has the highest latitude value (71.285446). This airport is located in Utqiaġvik, Alaska, the northernmost city in the United States. External research confirms that this airport is the northernmost publicly accessible airport in the country, validating the result obtained from the dataset.

QUESTION 2: What is the easternmost airport in the United States?

In [22]:
import pandas as pd

Upload airports.csv

In [23]:
airports = pd.read_csv("airports.csv")
airports.head()

,faa,name,lat,lon,alt,tz,dst,tzone
0,04G,Lansdowne Airport,41.130472,-80.619583,1044,-5,A,America/New_York
1,06A,Moton Field Municipal Airport,32.460572,-85.680028,264,-6,A,America/Chicago
2,06C,Schaumburg Regional,41.989341,-88.101243,801,-6,A,America/Chicago
3,06N,Randall Airport,41.431912,-74.391561,523,-5,A,America/New_York
4,09J,Jekyll Island Airport,31.074472,-81.427778,11,-5,A,America/New_York


Locate the Easternmost Airport in the United States

In [24]:
easternmost = us_airports.sort_values(by="lon", ascending=False)
easternmost.head()

,faa,name,lat,lon,alt,tz,dst,tzone
1290,SYA,Eareckson As,52.712275,174.113620,98,-9,A,America/Anchorage
942,MYF,Montgomery Field,32.475900,117.759000,17,8,A,Asia/Chongqing
396,DVT,Deer Valley Municipal Airport,33.411700,112.457000,1478,8,A,Asia/Chongqing
444,EPM,Eastport Municipal Airport,44.910111,-67.012694,45,-5,A,America/New_York
624,HUL,Houlton Intl,46.123083,-67.792056,489,-5,A,America/New_York


I determined that Eareckson Air Station (SYA) has the largest longitude value (174.113620). Since Alaska crosses the 180° meridian into the Eastern Hemisphere, this makes it the easternmost airport in the United States.

QUESTION 3: On February 12th, 2013, which New York area airport had the windiest weather?

In [ ]:
import pandas as pd

Load weather.csv

In [25]:
weather = pd.read_csv("weather.csv")
weather.head()

,origin,year,month,day,hour,temp,dewp,humid,wind_dir,wind_speed,wind_gust,precip,pressure,visib,time_hour
0,EWR,2013,1,1,1,39.02,26.06,59.37,270.0,10.35702,NaN,0.0,1012.0,10.0,2013-01-01T06:00:00Z
1,EWR,2013,1,1,2,39.02,26.96,61.63,250.0,8.05546,NaN,0.0,1012.3,10.0,2013-01-01T07:00:00Z
2,EWR,2013,1,1,3,39.02,28.04,64.43,240.0,11.50780,NaN,0.0,1012.5,10.0,2013-01-01T08:00:00Z
3,EWR,2013,1,1,4,39.92,28.04,62.21,250.0,12.65858,NaN,0.0,1012.2,10.0,2013-01-01T09:00:00Z
4,EWR,2013,1,1,5,39.02,28.04,64.43,260.0,12.65858,NaN,0.0,1011.9,10.0,2013-01-01T10:00:00Z


Filter for Februart 12, 2013

In [35]:
weather_02122013 = weather[
    (weather['year'] == 2013) &
    (weather['month'] == 2) &
    (weather['day'] == 12)
]

Determine highest wind speed and removing inaccurate wind speeds.

In [34]:
weather_clean = weather_02122013[weather_02122013['wind_speed'] < 200]
weather_clean.groupby('origin')['wind_speed'].max()

origin
EWR    21.86482
JFK    20.71404
LGA    23.01560
Name: wind_speed, dtype: float64

After filtering the dataset for February 12, 2013 and removing unrealistic outlier values, I compared the recorded wind speeds for the three New York area airports. LaGuardia Airport (LGA) had the highest wind speed at approximately 23.02 mph, making it the windiest airport on that date.